crates/provider-blobstore-nats/src/provider.rs

```diff
@@ -0,0 +1,397 @@
+    }
+
+    /// Attempt to connect to NATS url (with JWT credentials, if provided)
+    async fn connect(
+        &self,
+        cfg: NatsConnectionConfig,
+    ) -> anyhow::Result<async_nats::jetstream::context::Context> {
+        let mut opts = match (cfg.auth_jwt, cfg.auth_seed) {
+            (Some(jwt), Some(seed)) => {
+                let seed = KeyPair::from_seed(&seed).context("failed to parse seed key pair")?;
+                let seed = Arc::new(seed);
+                async_nats::ConnectOptions::with_jwt(jwt, move |nonce| {
+                    let seed = seed.clone();
+                    async move { seed.sign(&nonce).map_err(async_nats::AuthError::new) }
+                })
+            }
+            (None, None) => async_nats::ConnectOptions::default(),
+            _ => bail!("must provide both jwt and seed for jwt authentication"),
+        };
+        if let Some(tls_ca) = &cfg.tls_ca {
+            opts = add_tls_ca(tls_ca, opts)?;
+        } else if let Some(tls_ca_file) = &cfg.tls_ca_file {
+            let ca = fs::read_to_string(tls_ca_file)
+                .await
+                .context("failed to read TLS CA file")?;
+            opts = add_tls_ca(&ca, opts)?;
+        }
+
+        // Get the cluster_uri
+        let uri = cfg.cluster_uri.unwrap_or_default();
+
+        // Connect to the NATS server
+        let client = opts
+            .name("NATS Object Store Provider") // allow this to show up uniquely in a NATS connection list
+            .connect(uri.clone())
+            .await?;
+
+        // Get the JetStream context based on js_domain
+        let jetstream = if let Some(domain) = &cfg.js_domain {
+            async_nats::jetstream::with_domain(client.clone(), domain.clone())
+        } else {
+            async_nats::jetstream::new(client.clone())
+        };
+
+        debug!("opened NATS JetStream: {:?}", jetstream);
+        debug!("NATS Connection Configuration: {:?}", client);
+
+        // Return the handle to the opened NATS Object store
+        Ok(jetstream)
+    }
+
+    /// Helper function to lookup and return the NATS JetStream connection handle, and container storage
+    /// configuration, using the client component's context.
+    /// This ensures consistent implementation across all functions that need to get the NATS Blobstore.
+    pub(crate) async fn get_blobstore(
+        &self,
+        context: Option<Context>,
+    ) -> anyhow::Result<NatsBlobstore> {
+        if let Some((component_id, link_name)) =
+            context.as_ref().and_then(|Context { component, .. }| {
+                // Clone component and provide a placeholder default value for link_name if it's None,
+                // which will be replaced by the last link name set by the consumer.
+                let link_name = context
+                    .clone()
+                    .context("missing context")
+                    .unwrap()
+                    .link_name()
+                    .to_string();
```

Comment:
Are you able to pull the `link_name` from the `Context` object here to avoid the unwrap? I know it's safe because you're already in the code path where `context` is Some, but figured it may help readability.

Reply:
Agreed; I sometimes forget to take a step back and reevaluate the new improvements and the older code, as a whole; I simplified it.